--- a/modules/zip.c
+++ b/modules/zip.c
@@ -452,7 +452,7 @@
 	if(ulen<1 || ulen>1000000) goto done;
 
 	if(!is_compression_method_supported(cmprtype)) {
-		de_warn(c, "Finder attribute data: Unspported compression method: %d", (int)cmprtype);
+		de_warn(c, "Finder attribute data: Unsupported compression method: %d", (int)cmprtype);
 	}
 
 	// Decompress and decode the Finder attribute data
@@ -547,7 +547,7 @@
 	{ 0x4690 /*    */, "POSZIP 4690", NULL },
 	{ 0x4704 /*    */, "VM/CMS", NULL },
 	{ 0x470f /*    */, "MVS", NULL },
-	{ 0x4854 /* TH */, "Theos, old inofficial port", NULL },
+	{ 0x4854 /* TH */, "Theos, old unofficial port", NULL },
 	{ 0x4b46 /* FK */, "FWKCS MD5", NULL },
 	{ 0x4c41 /* AL */, "OS/2 access control list (text ACL)", NULL },
 	{ 0x4d49 /* IM */, "Info-ZIP OpenVMS", NULL },
--- a/modules/iptc.c
+++ b/modules/iptc.c
@@ -59,7 +59,7 @@
 	{ 1, 122, 0,      "ARM Version", handle_uint16 },
 	{ 2, 0,   0,      "Record Version", handle_uint16 },
 	{ 2, 3,   0x0001, "Object Type Reference", NULL },
-	{ 2, 4,   0x0001, "Object Atribute Reference", NULL },
+	{ 2, 4,   0x0001, "Object Attribute Reference", NULL },
 	{ 2, 5,   0x0001, "Object Name", NULL },
 	{ 2, 7,   0x0001, "Edit Status", NULL },
 	{ 2, 8,   0x0001, "Editorial Update", NULL },
@@ -119,7 +119,7 @@
 	{ 7, 10,  0,      "Size Mode", NULL },
 	{ 7, 20,  0,      "Max Subfile Size", NULL },
 	{ 7, 90,  0,      "ObjectData Size Announced", NULL },
-	{ 7, 95,  0,      "Maxium ObjectData Size", NULL },
+	{ 7, 95,  0,      "Maximum ObjectData Size", NULL },
 	{ 8, 10,  0,      "Subfile", NULL },
 	{ 9, 10,  0,      "Confirmed ObjectData Size", NULL }
 };
--- a/modules/abk.c
+++ b/modules/abk.c
@@ -715,7 +715,7 @@
 		dbuf_create_file_from_slice(c->infile, pos, c->infile->len-pos, "AmBs.abk", NULL, 0);
 	}
 	else {
-		de_dbg(c, "not extracting emtpy AmBs segment");
+		de_dbg(c, "not extracting empty AmBs segment");
 	}
 	de_dbg_indent(c, -1);
 
